### PR TITLE
chore: updated collector to v0.36.0

### PR DESCRIPTION
**Pull request by willarmiros**
Updates to upstream collector v0.36.0. I was pretty confident that I should be adding the newly separated components to `collector/lambdacomponens/go.mod` instead of `collector/go.mod`, but would appreciate if someone could double check.

### CODE REVIEW

File: collector/lambdacomponents/go.mod [hunk 1]
```diff
@@ -2,4 +2,13 @@ module github.com/open-telemetry/opentelemetry-lambda/collector/lambdacomponents
 
 go 1.16
```
Comment (anuraaga) [inline collector/lambdacomponents/go.mod:3]: Think this file's go.sum isn't updated yet

File: collector/lambdacomponents/go.mod [hunk 2]
```diff
@@ -2,4 +2,13 @@ module github.com/open-telemetry/opentelemetry-lambda/collector/lambdacomponents
...
 
-require go.opentelemetry.io/collector v0.33.0
+require (
+  go.opentelemetry.io/collector v0.36.0
+  github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter v0.36.0
```
Comment (anuraaga) [inline collector/lambdacomponents/go.mod:7]: Just wondering where did this list come from?

File: collector/lambdacomponents/go.mod (hunk 1 above)
Comment (willarmiros) [reply collector/lambdacomponents/go.mod:3]: Should be good now.

**Comment by willarmiros** [comment]
Fixed the breaking change with the new `Command` interface by referencing this PR: https://github.com/open-telemetry/opentelemetry-collector/pull/4074

File: collector/lambdacomponents/go.mod (hunk 2 above)
Comment (willarmiros) [reply collector/lambdacomponents/go.mod:7]: It's based on all the components here that were no longer in the core collector: https://github.com/open-telemetry/opentelemetry-lambda/blob/main/collector/lambdacomponents/default.go#L19-L30

File: collector/go.mod [hunk 3]
```diff
@@ -5,6 +5,21 @@ go 1.16
 replace github.com/open-telemetry/opentelemetry-lambda/collector/lambdacomponents => ./lambdacomponents
 
 require (
+	github.com/cespare/xxhash/v2 v2.1.2 // indirect
```
Comment (willarmiros) [inline collector/go.mod:8]: Is there any reason not to have these indirect imports? I noticed we were deleting them before.

Comment (anuraaga) [reply collector/go.mod:8]: We should keep whatever go adds

**Review by anuraaga — APPROVED** [review]

**Pull request merged**